test/2_test_fullpath_with_permit.ts

```diff
@@ -502,7 +502,7 @@ describe('Cashier and VoucherKernel', () => {
             constants.PROMISE_DEPOSITBU1,
             constants.QTY_10
           )
-        ).to.be.revertedWith(revertReasons.INVALID_VALIDITY_FROM);
+        ).to.be.revertedWith(revertReasons.INVALID_VALIDITY_TO);
```

Comment:
Revert reason does not match test description. But given that you made the tests for `VALID_FROM_MUST_BE_AT_LEAST_5_MINUTES_LESS_THAN_VALID_TO` (which have stricter requirements than this old test), this test here can be removed.

Same is true for the occurrences in the code below.